### PR TITLE
Allows to disable helm validation

### LINKED ISSUE #16552: Helm template with serviceMonitor enabled fails
## Bug report

**General Information**

- Cilium version: 1.10.0

Using `helm template` with `serviceMonitor` enabled fails due to the validation logic at https://github.com/cilium/cilium/blob/v1.10.0/install/kubernetes/cilium/templates/validate.yaml#L15.

The way I understand it, `helm install ...` and `helm upgrade ...` works as they speak with the k8s control plane and can check if the CRD is present, but `helm template ...` does not.

**How to reproduce the issue**

```shell
helm template cilium/cilium --version 1.10.0 --set prometheus.enabled=true --set prometheus.serviceMonitor.enabled=true
```
```shell
Error: template: cilium/templates/validate.yaml:16:9: executing "cilium/templates/validate.yaml" at <fail "Service Monitor requires monitoring.coreos.com/v1 CRDs. Please refer to https://github.com/prometheus-operator/prometheus-operator/blob/master/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml">: error calling fail: Service Monitor requires monitoring.coreos.com/v1 CRDs. Please refer to https://github.com/prometheus-operator/prometheus-operator/blob/master/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml

Use --debug flag to render out invalid YAML
```

### PR DESCRIPTION
**Pull request by farodin91**
Fixes #16552

Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

Fixes: #issue-number

```release-note
Allows to disable helm validation
```

### CODE REVIEW

File: install/kubernetes/cilium/values.yaml [hunk 1]
```diff
@@ -1798,3 +1798,6 @@ cgroup:
 # -- Configure whether to enable auto detect of terminating state for endpoints
 # in order to support graceful termination.
 enableK8sTerminatingEndpoint: true
+
+# -- Allows to disable the helm validation checks
```
Comment (qmonnet) [inline install/kubernetes/cilium/values.yaml:1802]:
```suggestion
# -- Allows to disable the Helm validation checks
```

File: install/kubernetes/cilium/values.yaml [hunk 2]
```diff
@@ -1798,3 +1798,6 @@ cgroup:
...
+enableHelmValidation: true
```
Comment (qmonnet) [inline install/kubernetes/cilium/values.yaml:1803]: Please preserve the line break at the end of the line.

**Review by qmonnet — CHANGES_REQUESTED** [review]
Thanks! The change looks good overall. It should be noted that it disables the whole Helm validation, when there was just one check that seemed to cause an issue in the original bug report, but I think it's OK.

One thing I would like is to keep a bit more context for the change. Could you please explain why this change is required, and why `helm template` doesn't work in your commit description? It's not fun to go and search for the PRs when `git blaming` a change to find the initial motivation. I'm also wondering whether we should extend the description of the new option to tell users _why_ it might be desirable to disable the checks.

**Review by gandro — COMMENTED** [review]
Thanks! Please also run `make -C install/kubernetes` to update the Helm Chart README.

**Comment by gandro** [comment]
While I'm not opposed to be able to opt-out from the validation, I would like to point out that the referenced issue is better solved by specifying the supported api-versions manually rather than disabling validation: https://github.com/cilium/cilium/issues/16552#issuecomment-998037683

**Review by christarazi — COMMENTED** [review]
Thanks for the PR. 

My opinion is that it seems like a clunky approach to workaround a problem when installing Helm charts. Ideally, we shouldn't need to provide an opt-out if the solution to the root issue is what @gandro suggested. @farodin91 Has that worked for you or are there more use cases for opting out of Helm validation?

**Comment by farodin91** [comment]
@gandro I'm currently playing around with ArgoCD. It seems the issue is solved with Argo CD 2.2. I'm okay with closing this PR. It could be helpful to document how to run helm template correctly.

**Comment by gandro** [comment]
> @gandro I'm currently playing around with ArgoCD. It seems the issue is solved with Argo CD 2.2. I'm okay with closing this PR. It could be helpful to document how to run helm template correctly.

I see. Looking at https://github.com/argoproj/argo-cd/issues/7291 indeed seems that Argo CD has remaining bugs around `--api-versions` depending on how it is used, which might justify adding this as a workaround for now.

**Comment by farodin91** [comment]
> > @gandro I'm currently playing around with ArgoCD. It seems the issue is solved with Argo CD 2.2. I'm okay with closing this PR. It could be helpful to document how to run helm template correctly.
> 
> I see. Looking at [argoproj/argo-cd#7291](https://github.com/argoproj/argo-cd/issues/7291) indeed seems that Argo CD has remaining bugs around `--api-versions` depending on how it is used, which might justify adding this as a workaround for now.

Validation issue of the servicemonitor is solved in argo 2.2. I would add a workaround only, if there another issue. 

I've you want that cilium wants this workaround, i'll update the PR to fix all comments.

**Comment by gandro** [comment]
> Validation issue of the servicemonitor is solved in argo 2.2. I would add a workaround only, if there another issue.

I'm not aware of any other issues related to validation at the moment. So yes, let's close this then. We can always re-open if we discover another issue.

@farodin91 Thanks a ton for looking into this and doing the work. Even if this will not be merged, your contribution and input was most welcome!

**Pull request closed without merging**